src/modules/peek/Peek.UI/FileQuery.cs

```diff
@@ -0,0 +1,170 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Peek.UI
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using CommunityToolkit.Mvvm.ComponentModel;
+    using Peek.Common.Models;
+    using Peek.UI.Helpers;
+
+    public partial class FileQuery : ObservableObject
+    {
+        private const int UninitializedItemIndex = -1;
+
+        public void Uninitialize()
```

Comment:
nit: instead of initialize and uninitialize, maybe Start() and Cancel()? I don't have a strong opinion, but init/uninit is very generic. Something that illustrates that the query is starting/querying the file system could add clarity.